Luanaruiz9 cfq data gen

luanaruiz9 opened the pull request:
Data generation script for CFQ iterative decoding data.

luanaruiz9 committed:
Created data_generation.py for CFQ

Generates iterative decoding data for the CFQ dataset.

luanaruiz9 committed:
Update data_generation.py

Added all val samples to test set (to be consistent with the seq2seq data) and fixed formatting.

luanaruiz9 committed:
Update data_generation.py

Added all val samples to test set (to be consistent with the seq2seq data) and fixed formatting.

luanaruiz9 committed:
Added csv_map.csv

Table connecting words (from questions) with the correspoding SPARQL commands (from queries).

santiaontanon-google reviewed (approved):
Thanks Luana, LGTM! Very curious to see if clause re-ordering has any effect on performance!

luanaruiz9 committed:
Update data_generation.py

Fixed bugs and changed clause reordering algorithm to score all clauses with a given command equally. Scores decrease with the position of the corresponding words in the input.

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+                             it_dec_test_filename_ops, simplify_output=False, random=True)
+                           
+                             
+if __name__ == '__main__':
```
Nit: since you use double quotes in this file
```suggestion
if __name__ == "__main__":
```

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+  for token in tokens:
+    if state == 0:
+      if token == "{":
+        prefix += token + " "
```
Nit: it may be more efficient/typical to append to a list of strings and join them at the end, but I doubt this is causing much of a speed problem so no worries.

(Comment also applies to the `generate_it_dec_examples` function below.)

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+  if not triplets:
+    return triplets
+  triplet = triplets[0]
+  if triplet[0]== "(":
```
```suggestion
  if triplet[0] == "(":
```

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+      ranking_list = []
+      for i, word in enumerate(words):
+        if word in csv_map.keys():
+            score = len(words) - i
+            this_ranking = np.zeros(len(clauses))
+            for j, clause in enumerate(clauses):
+                for value in csv_map[word]:
+                    if value + " " in clause and this_ranking[j] == 0:
+                        this_ranking[j] = score
+            ranking_list.append(this_ranking)
```
Nit: indentation is switching to 4 spaces instead of 2 here.

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+    for idx, clause in enumerate(clauses):
+      if idx == len(clauses) - 1:
+        if " ." in clause:
+            clauses[idx] = clause.replace(" .", "")  
```
Too much indentation here too.

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+      with open("cfq_map.csv", mode="r") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",")
+        for row in csv_reader:
+            if row[0] not in csv_map.keys():
+                csv_map[str(row[0])] = [str(row[1])]
+            else:
+                csv_map[str(row[0])].append(str(row[1]))
```
Too much indentation here too.

jtainslie commented on datasets/cfq/data_generation.py:
```diff
@@ -0,0 +1,341 @@
+    output_file.close()
+
+
+def generate_it_dec_examples(ds, split, filename_in, filename_out, filename_ops=None, 
```
The rest of the file has a mixture of 2-space and 4-space indentation.  The Google style guide recommends 2 spaces throughout.

jtainslie reviewed (approved):
Lgtm apart from some formatting nits.

luanaruiz9 committed:
Update datasets/cfq/data_generation.py

Co-authored-by: Joshua Ainslie <jtainslie@users.noreply.github.com>

luanaruiz9 committed:
Update datasets/cfq/data_generation.py

Co-authored-by: Joshua Ainslie <jtainslie@users.noreply.github.com>

luanaruiz9 committed:
Corrected indentation
